cagw_config_entry.go

```diff
@@ -5,8 +5,94 @@
 
 package main
 
-type CAGWConfigEntry struct {
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	"github.com/pkg/errors"
+)
+
+type CAGWEntry struct {
```

Comment:
Ideally the name of this struct should match the file name.

Reply:
Good catch. The struct names started to look too long when creating a composite struct for CAGWConfigEntry and CAGWConfigProfile. I shortened the name for this reason. But you have a valid point.